### PR TITLE
Support preventing default for text input

### PR DESCRIPTION
**Pull request by namse**
# What I did
- Remove text_input.update
- Support preventing default for text input's key down event

After #332

For hot-reloading or state restoration, I believe we have to implement text input without browser's support, using IME library directly someday.

### CODE REVIEW

File: namui/sample/text-input/src/lib.rs [hunk 1]
```diff
@@ -122,6 +124,12 @@ impl Entity for TextInputExample {
                 color: namui::Color::BLACK,
                 ..Default::default()
             },
+            event_handler: Some(text_input::EventHandler::new().on_key_down(|event| {
+                if event.code == namui::Code::Digit7 {
+                    namui::log!("7 key is pressed, prevent default");
+                    event.prevent_default();
+                }
+            })),
```
Comment (namse) [inline namui/sample/text-input/src/lib.rs:132]: This is the key of prevent-default usage

File: namui/src/namui/system/text_input/key_down.rs [hunk 2]
```diff
@@ -75,3 +89,54 @@ fn handle_selection_change(
         )
         .unwrap();
 }
+
+fn get_selection_on_keyboard_down(
```
Comment (namse) [inline namui/src/namui/system/text_input/key_down.rs:93]: below code has been moved from other file.

File: namui/src/namui/system/text_input/mouse_event.rs [hunk 3]
```diff
@@ -115,10 +116,131 @@ fn update_focus_with_mouse_movement(
...
-
     let event = text_input::Event::Focus {
-        id: custom_data.text_input.id.clone(),
+        id: custom_data.id.clone(),
         selection,
     };
     crate::event::send(event);
 }
+
+fn get_selection_on_mouse_movement(
```
Comment (namse) [inline namui/src/namui/system/text_input/mouse_event.rs:126]: below code has been moved from other file.

File: namui/src/namui/render/text_input/mod.rs [hunk 4]
```diff
@@ -28,39 +26,60 @@ pub struct Props {
...
+        f.debug_struct("EventHandler")
+            .field("on_key_down", &self.on_key_down.is_some())
+            .finish()
+    }
 }
+
+pub struct KeyDownEvent {
+    pub code: Code,
+    pub(crate) is_prevented_default: Arc<AtomicBool>,
+    pub is_composing: bool,
```
Comment (namse) [inline namui/src/namui/render/text_input/mod.rs:58]: This is for https://github.com/vuejs/vue/issues/10277

**Review by bigfoodK — APPROVED** [review]

**Pull request merged**